Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,521 @@
+
+                var commandName = command.CommandType == CommandType.StoredProcedure
+                    ? command.CommandText
+                    : string.Empty;
+
+                dependencyName = string.IsNullOrEmpty(commandName)
+                    ? string.Join(" | ", command.Connection.DataSource, command.Connection.Database)
+                    : string.Join(" | ", command.Connection.DataSource, command.Connection.Database, commandName);
+            }
+
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            var telemetry = new DependencyTelemetry()
+            {
+                Id = Convert.ToString(operationId, CultureInfo.InvariantCulture),
+                Name = dependencyName,
+                Type = RemoteDependencyConstants.SQL,
+                Target = target,
+                Data = command.CommandText,
+                Duration = duration,
+                Timestamp = DateTimeOffset.UtcNow - duration,
+                Success = true
+            };
+
+            InitializeTelemetry(telemetry, operationId);
+            
+            this.client.Track(telemetry);
+        }
+
+        private void OnCommandErrorEvent(
+            Guid operationId,
+            string operation,
+            Guid connectionId,
+            Exception exception,
+            long endTimestamp)
+        {
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            var telemetry = new ExceptionTelemetry(exception)
+            {
+                Message = exception.Message,
+                Timestamp = DateTimeOffset.UtcNow - duration,
+                SeverityLevel = SeverityLevel.Critical
+            };
+
+            InitializeTelemetry(telemetry, operationId);
+            CorrelateErrorTelemetry(telemetry, operationId);
+
+            this.client.Track(telemetry);
+        }
+
+        private void OnAfterConnectionEvent(
+            Guid operationId, 
+            string operation,
+            Guid connectionId,
+            SqlConnection connection, 
+            IDictionary statistics, 
+            long endTimestamp)
+        {
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+            
+            var telemetry = new DependencyTelemetry()
+            {
+                Id = Convert.ToString(operationId, CultureInfo.InvariantCulture),
+                Name = string.Join(" | ", connection.DataSource, connection.Database, operation),
+                Type = RemoteDependencyConstants.SQL,
+                Target = string.Join(" | ", connection.DataSource, connection.Database),
+                Data = operation,
+                Duration = duration,
+                Timestamp = DateTimeOffset.UtcNow - duration,
+                Success = true
+            };
+
+            InitializeTelemetry(telemetry, operationId);
+
+            this.client.Track(telemetry);
+        }
+
+        private void OnErrorConnectionEvent(
```

Comment:
Shouldn't be. [Here](https://github.com/dotnet/corefx/blob/83b606b5debad20536a581c425f66f4e9c396ed1/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs#L809)is an example of the usage pattern they employ in SqlClient.